internal/pipeline/step.go

```diff
@@ -29,6 +30,29 @@ type Step interface {
 	selfInterpolater
 }
 
+type ScalarStep string
```

Comment:
My feelings about the types here: I kinda want for it so that the types are easy to assemble in order to produce pipelines with Go. "ScalarStep" describes "a shape of data" more than "a type of step", and also overlaps semantic meaning with WaitStep and InputStep. But it's not too hard to use.

Reply:
yeah, i had this thought as well... the other idea i had was something like 
```Go
type WaitStep struct {
	Scalar string `yaml:"-"`
	RemainingFields map[string]any `yaml:",inline"`
}
```
and then have the marshal return the value of `scalar` if it's defined, and a regular marshal if not. is that something that feels better to you?

Reply:
Yeah sort of. I feel like if someone wants to write `WaitStep{}` it should marshal as `"wait"`, because that's the kind of step it is.

Reply:
this is updated in https://github.com/buildkite/agent/pull/2253/commits/a9621be2e078f366c3acd5dd1a266277115dcbf1 - would love your thoughts!